flink-connectors/flink-connector-jdbc/src/main/java/org/apache/flink/connector/jdbc/dialect/AbstractDialect.java

```diff
@@ -18,60 +18,65 @@
 
 package org.apache.flink.connector.jdbc.dialect;
 
-import org.apache.flink.table.api.TableSchema;
+import org.apache.flink.annotation.PublicEvolving;
 import org.apache.flink.table.api.ValidationException;
 import org.apache.flink.table.types.DataType;
 import org.apache.flink.table.types.logical.DecimalType;
 import org.apache.flink.table.types.logical.LogicalTypeRoot;
+import org.apache.flink.table.types.logical.RowType;
 import org.apache.flink.table.types.logical.TimestampType;
 import org.apache.flink.table.types.logical.VarBinaryType;
 
 import java.util.List;
 
-abstract class AbstractDialect implements JdbcDialect {
+/** Base class that handles common validation of JDBC data types. */
+@PublicEvolving
```

Comment:
Should we really make this public already? If so, what is the contract that we guarantee here (other than `JdbcDialect`'s)? This deserves better documentation, I think.

I'm also not sure the design of this class is great yet. For example implementers are forced to specifiy min/max precisions, maybe that should be optional? Also, shouldn't an implementation list supported types rather than unsupported ones?

Reply:
yep. basically rewrote this class.